File: .claude/skills/py-cli-beautifier/scripts/colorize_output.py
#!/usr/bin/env python3
"""
colorize_output.py - Functions for applying consistent color schemes to Python CLI output
"""

try:
    from rich.console import Console
    from rich.text import Text
    from rich import print as rich_print
    RICH_AVAILABLE = True
except ImportError:
    RICH_AVAILABLE = False

try:
    from colorama import init, Fore, Back, Style
    init()  # Initialize colorama
    COLORAMA_AVAILABLE = True
except ImportError:
    COLORAMA_AVAILABLE = False


def colorize_rich(message, color="white"):
    """Colorize message using rich library"""
    if not RICH_AVAILABLE:
        raise ImportError("rich library is not available")

    return Text(message, style=color)


def colorize_colorama(message, color="white"):
    """Colorize message using colorama library"""
    if not COLORAMA_AVAILABLE:
        raise ImportError("colorama library is not available")

    color_map = {
        "red": Fore.RED,
        "green": Fore.GREEN,
        "yellow": Fore.YELLOW,
        "blue": Fore.BLUE,
        "magenta": Fore.MAGENTA,
        "cyan": Fore.CYAN,
        "white": Fore.WHITE,
        "reset": Style.RESET_ALL
    }

    color_code = color_map.get(color.lower(), Fore.WHITE)
    return f"{color_code}{message}{Style.RESET_ALL}"


def format_message_rich(message_type, message):
    """Format a message with color and icon using rich"""
    if not RICH_AVAILABLE:
        raise ImportError("rich library is not available")

    icons = {
        "success": "✅",
        "error": "❌",
        "warning": "⚠️",
        "info": "ℹ️",
        "debug": "🐛",
        "highlight": "✨",
        "title": "🎯"
    }

    colors = {
        "success": "green",
        "error": "red",
        "warning": "yellow",
        "info": "blue",
        "debug": "grey62",
        "highlight": "cyan",
        "title": "magenta"
    }

    icon = icons.get(message_type, "🔹")
    color = colors.get(message_type, "white")

    text = Text()
    text.append(icon + " ", style=color)
    text.append(message, style=color)

    return text


def format_message_colorama(message_type, message):
    """Format a message with color and icon using colorama"""
    if not COLORAMA_AVAILABLE:
        raise ImportError("colorama library is not available")

    icons = {
        "success": "✅",
        "error": "❌",
        "warning": "⚠️",
        "info": "ℹ️",
        "debug": "🐛",
        "highlight": "✨",
        "title": "🎯"
    }

    colors = {
        "success": Fore.GREEN,
        "error": Fore.RED,
        "warning": Fore.YELLOW,
        "info": Fore.BLUE,
        "debug": Fore.LIGHTBLACK_EX,
        "highlight": Fore.CYAN,
        "title": Fore.MAGENTA
    }

    icon = icons.get(message_type, "🔹")
    color_code = colors.get(message_type, Fore.WHITE)

    return f"{color_code}{icon} {message}{Style.RESET_ALL}"


def create_status_banner_rich(title, status, status_type="info"):
    """Create a status banner using rich"""
    if not RICH_AVAILABLE:
        raise ImportError("rich library is not available")

    status_colors = {
        "success": "green",
        "error": "red",
        "warning": "yellow",
        "info": "blue"
    }

    color = status_colors.get(status_type, "white")
    status_text = Text(status, style=f"bold {color}")

    from rich.panel import Panel
    return Panel(f"Name: {title}\nStatus: {status}", title="Application Status")


def create_status_banner_simple(title, status, status_type="info"):
    """Create a simple status banner without external libraries"""
    banner = f"""
┌─ APPLICATION STATUS ──────────────────────────────────┐
│  Name: {title:<43} │
│  Status: {status:<40} │
└────────────────────────────────────────────────────────┘
"""
    return banner


# Example usage
if __name__ == "__main__":
    if RICH_AVAILABLE:
        print("Rich formatted messages:")
        rich_print(format_message_rich("success", "Operation completed successfully"))
        rich_print(format_message_rich("error", "An error occurred during processing"))
        rich_print(format_message_rich("warning", "This is a warning message"))
        rich_print(format_message_rich("info", "Here is some important information"))

        print("\nRich status banner:")
        from rich.console import Console
        console = Console()
        console.print(create_status_banner_rich("My App", "Running", "success"))

    if COLORAMA_AVAILABLE:
        print("\nColorama formatted messages:")
        print(format_message_colorama("success", "Operation completed successfully"))
        print(format_message_colorama("error", "An error occurred during processing"))
        print(format_message_colorama("warning", "This is a warning message"))
        print(format_message_colorama("info", "Here is some important information"))

    print("\nSimple status banner:")
    print(create_status_banner_simple("My App", "Running"))

File: .claude/skills/js-cli-beautifier/scripts/colorize-output.js
#!/usr/bin/env node

// colorize-output.js - Functions for applying consistent color schemes to CLI output

// Dynamically require chalk to handle cases where it might not be available
let chalk;
try {
  chalk = require('chalk');
} catch (e) {
  // Fallback if chalk is not available
  chalk = {
    green: (str) => str,
    red: (str) => str,
    yellow: (str) => str,
    blue: (str) => str,
    gray: (str) => str,
    cyan: (str) => str,
    magenta: (str) => str,
    white: (str) => str,
    bold: (str) => str
  };
  // Create compound functions for the expected combinations
  chalk.green.bold = (str) => str;
  chalk.red.bold = (str) => str;
  chalk.yellow.bold = (str) => str;
  chalk.blue.bold = (str) => str;
  chalk.cyan.bold = (str) => str;
  chalk.magenta.bold = (str) => str;
  chalk.magenta.bold.underline = (str) => str;
}

/**
 * Color schemes for different types of messages
 */
const colorSchemes = {
  success: chalk.green.bold,
  error: chalk.red.bold,
  warning: chalk.yellow.bold,
  info: chalk.blue.bold,
  debug: chalk.gray,
  highlight: chalk.cyan.bold,
  title: chalk.magenta.bold.underline,
  subtle: chalk.gray
};

/**
 * Applies consistent color formatting to messages
 * @param {string} type - Type of message (success, error, warning, info, etc.)
 * @param {string} message - The message to colorize
 * @returns {string} Colorized message
 */
function colorize(type, message) {
  const colorFn = colorSchemes[type] || chalk.white;
  return colorFn(message);
}

/**
 * Creates a formatted message with an icon
 * @param {string} type - Type of message
 * @param {string} message - The message content
 * @returns {string} Formatted message with icon
 */
function formatMessage(type, message) {
  const icons = {
    success: '✅',
    error: '❌',
    warning: '⚠️',
    info: 'ℹ️',
    debug: '🐛',
    highlight: '✨',
    title: '🎯',
    subtle: '🔸'
  };

  const icon = icons[type] || '🔹';
  const colorFn = colorSchemes[type] || chalk.white;

  return `${icon} ${colorFn(message)}`;
}

/**
 * Creates a status banner with consistent styling
 * @param {string} title - Banner title
 * @param {string} status - Status message
 * @param {string} color - Color for the status
 * @returns {string} Formatted banner
 */
function createStatusBanner(title, status, color = 'info') {
  const colorFn = colorSchemes[color];
  const statusText = colorFn ? colorFn(status) : status;

  const banner = `
┌─────────────────────────────────────────────────────────┐
│  ${chalk.bold(title)}${' '.repeat(48 - title.length)} │
│  Status: ${statusText}${' '.repeat(39 - status.length)} │
└─────────────────────────────────────────────────────────┘
`;

  return banner;
}

/**
 * Formats a list of items with consistent styling
 * @param {Array<string>} items - List of items to format
 * @param {string} title - Optional title for the list
 * @returns {string} Formatted list
 */
function formatList(items, title = '') {
  if (title) {
    console.log(chalk.bold.blue(`\n${title}\n`));
  }

  items.forEach((item, index) => {
    console.log(`${chalk.cyan(`  ${index + 1}.`)} ${item}`);
  });
}

// Example usage
if (require.main === module) {
  console.log(formatMessage('title', 'CLI Beautification Examples'));
  console.log('');
  console.log(formatMessage('success', 'Operation completed successfully'));
  console.log(formatMessage('error', 'An error occurred'));
  console.log(formatMessage('warning', 'This is a warning message'));
  console.log(formatMessage('info', 'Here is some information'));
  console.log(formatMessage('debug', 'Debug information'));

  console.log(createStatusBanner('My CLI Application', 'Running', 'success'));

  formatList([
    'Task 1: Initialize project',
    'Task 2: Install dependencies',
    'Task 3: Build application',
    'Task 4: Run tests'
  ], 'Build Process');
}

module.exports = {
  colorize,
  formatMessage,
  createStatusBanner,
  formatList,
  colorSchemes
};